scylla/src/transport/load_balancing/default.rs

```diff
@@ -272,6 +306,24 @@ impl DefaultPolicy {
             .filter(predicate)
     }
 
+    fn pick_local_rack_replica<'a>(
+        &'a self,
+        ts: &TokenWithStrategy<'a>,
+        should_be_local: bool,
+        predicate: &impl Fn(&NodeRef<'a>) -> bool,
+        cluster: &'a ClusterData,
+    ) -> Option<NodeRef<'a>> {
+        if let Some(rack) = &self.preferred_rack {
+            return self
+                .nonfiltered_replica_set(ts, should_be_local, cluster)
+                .choose_filtered(&mut thread_rng(), |node| {
+                    predicate(&node) && node.rack.as_deref() == Some(rack)
+                });
+        }
+
+        None
+    }
+
```

Comment:
In my opinion, creating a distinct function for selecting local rack replicas is unnecessary. Instead, the desired outcome can be accomplished by utilizing a suitable predicate with the `pick_replicas` function.

Furthermore, it appears that the modifications introduced by rebasing were not taken into account. While `pick_replica` has the ability to select from an unshuffled sequence, this capability is not present in `pick_local_rack_replica`, which results in a lack of support for the `enable_shuffling_replicas(false)` option.

Overall, I believe this reinforces the argument against replicating the functionality of `pick_replica` and instead utilizing a custom predicate to choose local rack replicas.